command/command.go

```diff
@@ -203,3 +198,48 @@ func (s *shiftUpCommand) Execute() {
 func (s *shiftDownCommand) Execute() {
 	s.ShiftDown()
 }
+
+// Manages interactions between Player and Transmission
+type shiftController struct {
+	*transmission
+	*shiftDownCommand
+	*shiftUpCommand
+}
+
+// Returns a new shiftController with pointers to a transmission, shiftUpCommand and shiftDownCommand.
+// Things to consider: once we have multiple players each shifting their own transmission, should those interactions all be handled by a single shiftController, or should each player/transmission pair get it's own shiftController?
+func newShiftController() *shiftController {
+	t := newTransmission()
+	d := newShiftDownCommand(t)
+	u := newShiftUpCommand(t)
+	return &shiftController{t, d, u}
+}
+
+// Handles shifting a transmission up for a player
+func (s *shiftController) ShiftUpAction(c *CommandsList) {
+	s.shiftUpCommand.Execute()
+
+	// don't know if this is the best approach, but I don't think the alternative - returning the shiftCommand as a "success" value and having the Player's shiftUp method push that onto the slice - is really any better
+	c.PushCommand(s.shiftUpCommand)
+}
+
+// Handles shifting a transmission down for a player
+func (s *shiftController) ShiftDownAction(c *CommandsList) {
+	s.shiftDownCommand.Execute()
+	c.PushCommand(s.shiftDownCommand)
+}
+
+// Handles undoing the last shift action for a player
+func (s *shiftController) UndoLastAction() {
```

Comment:
I know this is unimplemented but it should probably return an error if you can't undo the last action. In Go, you can return an error as you do below with an implicit error, or you can return nil (error is an interface and therefore is a pointer type)